### PR TITLE
Intellij Dev Guide

### PR DESCRIPTION
**Pull request by jaycarlton**
First batch of things everyone should know (or at least know of) in IntelliJ.

### CODE REVIEW

File: docs/dev-guides/intellij.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
+# IntelliJ for Power Users
+IntelliJ has many secrets, and the easiest way to discover them is on accident. However, it's worth
+compiling some of the most useful ones in one place, so that one can benefit more quickly from them.
+All keyboard shortcuts are  9-5for the MacOS scheme.
+
+## Editing
+### Autocomplete
```
Comment (jaycarlton) [inline docs/dev-guides/intellij.md:26]: I forgot rectangular edit mode.

File: docs/dev-guides/intellij.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
+# IntelliJ for Power Users
+IntelliJ has many secrets, and the easiest way to discover them is on accident. However, it's worth
+compiling some of the most useful ones in one place, so that one can benefit more quickly from them.
+All keyboard shortcuts are  9-5for the MacOS scheme.
```
Comment (zloery) [inline docs/dev-guides/intellij.md]: "  9-5"?

File: docs/dev-guides/intellij.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+## Editing
+### Autocomplete
+When typing symbols, auto-completion is available. Type tab then Enter to accept
+the highlighted term, or use up and down keys to choose another one.
+![Autocomplete](./images/intellij/autocomplete.png)
+
+### Generate
+![Generate](./images/intellij/generate.png)
+Ctrl+N brings up a generate dialog, which can auto-generate
```
Comment (zloery) [inline docs/dev-guides/intellij.md:35]: if this is for the macOS scheme, isn't this cmd+N?

File: docs/dev-guides/intellij.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+#### Introduce Method
+To introduce a method, select an expression or group of expressions with a well-defined set of inputs
+and zero or one output. Not all selections are valid methods, naturally, and occasionally it's necessarily
+to refactor slightly so that the tool gets the right result. Frequently, it considers values to be constant
+that are desired as parameters. Those can be fixed with the Change Signature refactoring.
+
+#### Extract Interface
+A class with public methods can define an interface for those methods
+and then override them with the Extract Interface refactoring. This can
+be useful for declaring Sprinig beans that depend only on the interface,
```
Comment (zloery) [inline docs/dev-guides/intellij.md]: nit: Spring

File: docs/dev-guides/intellij.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+### Recent Files
+Cmd+E opens the Recent Files dialog. This list can be refined by using initial letter searches of filenames.
+It's especially useful when flipping to the most recently used file, since that's the file that's selected
+initially. On the left side is a list of recently active tool panes for quick navigation.
+![img](./images/intellij/recent_files_list.png)
+![img](./images/intellij/recent_files_selection.png)
+### Goto Line
+Cmd+G brings up an input box taking a line number to visit. A column can also be provided after a colon.
+
+![goto](./images/intellij/)
```
Comment (zloery) [inline docs/dev-guides/intellij.md]: broken link

File: docs/dev-guides/intellij.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+## Database
+
+## Equipment
+### Keyboard
+It's very helpful to have a full-size keyboard with all of Ctrl, Cmd, and Option on both sides. Having
+a missing control key on the right side is like trying to play a piano where all the C sharp keys are missing
+on the right-hand side. It's just painful.
+
+Additionally, it's very helpful to have a mechanical keyboard with a crisp stroke. This gives confidence
```
Comment (zloery) [inline docs/dev-guides/intellij.md:235]: todo: remove this advice when everyone is back in an office :^)

**Review by zloery — APPROVED** [review]
nits and proofreading, otherwise good.
Thanks for this!

File: docs/dev-guides/intellij.md (hunk 3 above)
Comment (jaycarlton) [reply docs/dev-guides/intellij.md:35]: Control is still a key for Mac keyboards, though command and option do most of the heavy lifting. Does it not work for you?

Comment (zloery) [reply docs/dev-guides/intellij.md:35]: yeah, I meant that on my mac keyboard cmd+N brings up the generate dialog, ctrl+N does something else

**Comment by jaycarlton** [comment]
Hmm. Cmd+N is Class search for me.

I'll update it.

-Jay

**Pull request merged**